pkg/metrics/metrics.go

```diff
@@ -14,49 +14,59 @@
  * limitations under the License.                                           *
  ***************************************************************************/
 
-// Package event //
-package event
-
-// Metrics is the interface for event processor
-type Metrics interface {
-	SetQueueSize(queueSize int)
-	IncrSuccessFlushCount()
-	IncrFailFlushCount()
-	IncrRetryFlushCount()
+// Package metrics //
+package metrics
+
+import (
+	"sync"
+	"time"
+)
+
+// GenericMetrics provides the interface for the metrics
+type GenericMetrics interface {
+	Inc(key string)
+	Set(key string, val int64)
+	Get(key string) int64
 }
 
-// DefaultMetrics stores the actual metrics
-type DefaultMetrics struct {
-	QueueSize         int
-	SuccessFlushCount int64
-	FailFlushCount    int64
-	RetryFlushCount   int64
+// Metrics contains default metrics
+type Metrics struct {
```

Comment:
I think the default implementation should be no-op. As it stands now this is not accessible and will only hurt the application performance. Providing the injection interface should be sufficient for consumers to implement. In the future we can provide an out-of-the-box implementation, but it should probably be optional.

Reply:
we need to have some implementation regardless. If we don't care to collect metrics on go-sdk yet then I can provide minimal implementation that does nothing (or check every time for nil interface) Also, I don't think injection interface will work really nicely with expvar variables as you need to define them globally - that means in sidedoor and pass them to go-sdk.